docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc

```diff
@@ -0,0 +1,273 @@
+
+`snapshot`::
+(string)
+Name of the snapshot.
+
+`uuid`::
+(string)
+Universally unique identifier (UUID) of the snapshot.
+
+`state`::
++
+--
+(string)
+Indicates the current snapshot state.
+
+.Values for `state`
+[%collapsible%open]
+====
+`FAILED`::
+  The snapshot finished with an error and failed to store any data.
+
+`IN_PROGRESS`::
+  The snapshot is currently running.
+
+`PARTIAL`::
+  The global cluster state was stored, but data of at least one shard was not stored successfully.
+  The <<get-snapshot-api-response-failures,`failures`>> section of the response contains more detailed information about shards
+  that were not processed correctly.
+
+`SUCCESS`::
+  The snapshot finished and all shards were stored successfully.
+====
+--
+
+`include_global_state`::
+(boolean)
+Indicates whether the current cluster state is included in the snapshot.
+
+[[get-snapshot-status-shards-stats]]
+`shards_stats`::
+(object)
+Contains a count of shards in the snapshot.
++
+.Properties of `shards_stats`
+[%collapsible%open]
+====
+`initializing`::
+(integer)
+Number of shards that are still initializing.
+
+`started`::
+(integer)
+Number of shards that have started but not are not finalized.
+
+`finalizing`::
+(integer)
+Number of shards that are finalizing but are not done.
+
+`done`::
+(integer)
+Number of shards that initialized, started, and finalized successfully.
+
+`failed`::
+(integer)
+Number of shards that failed to be included in the snapshot.
+
+`total`::
+(integer)
+Total number of shards included in the snapshot.
+====
+
+[[get-snapshot-status-stats]]
+`stats`::
+(object)
+Provides details on the number (`file_count`) and size (`size_in_bytes`) of files included in the snapshot.
++
+.Properties of `stats`
+[%collapsible%open]
+====
+`incremental`::
+(object)
+Number and size of files that are not already in the repository and need to be copied as part of the incremental snapshot.
+
+`processed`::
+(object)
+Number and size of files that are still being copied.
+
+`total`::
+(object)
+Total number and size of files that are referenced by the snapshot.
+
+`start_time_in_millis`::
+(long)
+The time, in milliseconds, when the snapshot creation process started.
+
+`time_in_millis`::
+(long)
+The total time, in milliseconds, that
+====
+
+`indices`::
+(list of objects)
+List of objects containing information about the
+indices included in the snapshot.
++
+.Properties of `indices`
+[%collapsible%open]
+====
+`shards_stats`::
+See <<get-snapshot-status-shards-stats,`shards_stats`>>.
+
+`stats`::
+See <<get-snapshot-status-stats,`stats`>>.
+
+`shards`::
+(list of objects)
+List of objects containing information about the
+shards that include the snapshot.
++
+.Properties of `shards`
+[%collapsible%open]
+=====
+`stage`::
+(string)
+Indicates the current state of the shards that include the snapshot.
++
+.Properties of `stage`
+[%collapsible%open]
+======
+`DONE`::
+Number of shards containing a completed copy of the snapshot.
```

Comment:
This is confusing (same for the other stage explanations here). i think something along the lines of the following would be better:

> Number of shards in the snapshot that were successfully stored in the repository.